File: day1/puzzle1.py
import re


def main():
    exp = re.compile("[0-9]")
    sum = 0
    with open("puzzle_input.txt", "r") as f:
        for line in f:
            digits = exp.findall(line)
            sum += int(digits[0]) * 10 + int(digits[-1])
    print(f"sum = {sum}")


if __name__ == "__main__":
    main()

File: day1/puzzle1.dart
import 'dart:convert';
import 'dart:io';

void main() async {
  final lines = utf8.decoder
      .bind(File("puzzle_input.txt").openRead())
      .transform(const LineSplitter());
  final regex = RegExp(r'(\d)');
  int sum = 0;

  await for (final line in lines) {
    final matches = [
      for (final match in regex.allMatches(line)) int.parse(match.group(0)!)
    ];
    sum += matches.first * 10 + matches.last;
  }
  print('sum = $sum');
}
